partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/TypeLattice.kt

```diff
@@ -266,41 +266,57 @@ internal class TypeLattice private constructor(
                 SYMBOL to explicit(),
             )
             graph[DECIMAL] = relationships(
-                INT8 to explicit(),
-                INT16 to explicit(),
-                INT32 to explicit(),
-                INT64 to explicit(),
                 BOOL to explicit(),
-                DECIMAL to explicit(),
-                DECIMAL_ARBITRARY to explicit(),
+                INT8 to unsafe(),
+                INT16 to unsafe(),
+                INT32 to unsafe(),
+                INT64 to unsafe(),
+                INT to unsafe(),
+                DECIMAL to coercion(),
```

Comment:
This is an interesting one. It is probably fine now as this gets converted to StaticType.Decimal anyway. But when we have the notion of precision and scale, this will not be true.